src/main/java/com/uber/kafkaSpraynozzle/KafkaSpraynozzle.java

```diff
@@ -42,13 +47,40 @@ public static void main(String[] args) {
             System.err.println(e.getMessage());
             return;
         }
+
+        String zk = spraynozzleArgs.getZk();
+        Boolean spraynozzleHA = spraynozzleArgs.getIsHighlyAvailable();
+        boolean buffering = spraynozzleArgs.getBuffering();
+        String hostName = getHostNameFromInetAddress();
+        if (isStringEmpty(hostName)) {
+            hostName = getHostNameFromFile();
+        }
+
+        if (spraynozzleHA) {
+            System.out.println("Performing leader election through zookeeper and picking leader that will proceed.");
+            //use same zk as kafka
+            String zkLeaderLatchFolderPath = "/consumers/kafka_spraynozzle_leader_latch";
+            String zkLeaderElectionFolderPath = "/consumers/kafka_spraynozzle_leader_elections";
+            SpraynozzleLeaderLatch curatorClient = new SpraynozzleLeaderLatch(zk, zkLeaderLatchFolderPath, zkLeaderElectionFolderPath, "spraynozzle-" + hostName);
```

Comment:
My comment seems to have disappeared. We'll need to include the topic name and cleaned url for the leader election, as well, to allow multiple rings of spraynozzles to coexist on the same hosts.

This is probably also true of the leader latch and leader election paths above. Multiple simultaneous clusters could be connected to the same zookeeper, so they need to be distinct.